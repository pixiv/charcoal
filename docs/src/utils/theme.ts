import styled from 'styled-components'
import { createTheme } from '@charcoal-ui/styled'

export const theme = createTheme(styled)
