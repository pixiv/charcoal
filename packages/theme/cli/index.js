/* eslint-disable @typescript-eslint/no-var-requires */

const { light, dark } = require('@charcoal-ui/theme')
const fs = require('fs')
const { parseToRgb } = require('polished')

/**
 * transform color string to [0, 1] clamped value of color object
 *
 * @param { import('@charcoal-ui/foundation').Material } material
 */
function parseMaterial(material) {
  const { red, green, blue, ...rest } = parseToRgb(material)
  const alpha = 'alpha' in rest ? rest.alpha : 1
  return { red: red / 255, green: green / 255, blue: blue / 255, alpha }
}

const deprecatedColor = ['callToAction']

const monolith = Object.fromEntries(
  Object.entries({ dark, light }).map(([id, original]) => {
    const transformed = {
      ...original,
      color: Object.fromEntries(
        Object.entries(original.color)
          .filter(([key]) => !deprecatedColor.includes(key))
          .map(([key, material]) => [key, parseMaterial(material)])
      ),
      gradientColor: Object.fromEntries(
        Object.entries(original.gradientColor).map(([key, def]) => [
          key,
          def.map((v) => ({ ...v, color: parseMaterial(v.color) })),
        ])
      ),
      effect: Object.fromEntries(
        Object.entries(original.effect).map(([key, def]) => [
          key,
          def.type !== 'opacity'
            ? { ...def, color: parseMaterial(def.color) }
            : def,
        ])
      ),
      border: Object.fromEntries(
        Object.entries(original.border).map(([key, def]) => [
          key,
          { ...def, color: parseMaterial(def.color) },
        ])
      ),
      outline: Object.fromEntries(
        Object.entries(original.border).map(([key, def]) => [
          key,
          { ...def, color: parseMaterial(def.color) },
        ])
      ),
    }
    return [id, transformed]
  })
)

fs.writeFileSync('theme.json', JSON.stringify(monolith, null, 2))
