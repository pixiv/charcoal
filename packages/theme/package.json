{
  "name": "@charcoal-ui/theme",
  "version": "2.5.0",
  "license": "Apache-2.0",
  "main": "./dist/index.cjs.js",
  "module": "./dist/index.esm.js",
  "exports": {
    "require": "./dist/index.cjs.js",
    "default": "./dist/index.esm.js"
  },
  "types": "./dist/index.d.ts",
  "sideEffects": false,
  "scripts": {
    "build": "npm-run-all --print-label --parallel 'build:*' --sequential serialize",
    "build:bundle": "FORCE_COLOR=1 tsup",
    "build:dts": "tsc --project tsconfig.build.json --pretty --emitDeclarationOnly",
    "serialize": "node cli/index.js",
    "typecheck": "run-p --print-label 'typecheck:*'",
    "typecheck:main": "tsc --project tsconfig.build.json --pretty --noEmit",
    "typecheck:cli": "tsc --project cli/tsconfig.build.json --noEmit",
    "clean": "rimraf dist .tsbuildinfo"
  },
  "devDependencies": {
    "npm-run-all": "^4.1.5",
    "rimraf": "^3.0.2",
    "tsup": "^6.5.0",
    "typescript": "^4.9.5"
  },
  "dependencies": {
    "@charcoal-ui/foundation": "^2.5.0",
    "@charcoal-ui/utils": "^2.5.0",
    "polished": "^4.1.4"
  },
  "files": [
    "src",
    "dist",
    "theme.json"
  ],
  "publishConfig": {
    "access": "public"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/pixiv/charcoal.git",
    "directory": "packages/theme"
  }
}
