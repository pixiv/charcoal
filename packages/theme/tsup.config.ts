import { defineConfig } from 'tsup'

export default defineConfig({
  entry: {
    index: 'src/index.ts',
    'unstable-token-object/index': 'src/unstable-token-object/index.ts',
  },
  format: ['esm', 'cjs'],
  outExtension({ format }) {
    return {
      js: `.${format}.js`,
    }
  },
  target: 'esnext',
  sourcemap: true,
  clean: true,
  dts: true,
})
