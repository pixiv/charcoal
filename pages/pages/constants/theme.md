# `@charcoal-ui/theme`

charcoal 全体で制約されるテーマ構造の型を提供します。

## インストール

npm

```bash
npm i --save-dev @charcoal-ui/theme
```

yarn

```bash
yarn add -D @charcoal-ui/theme
```
