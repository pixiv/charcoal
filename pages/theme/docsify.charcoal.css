:root {
  --color-brand: #0096fa;
  --color-surface3: #0000000a;
  --color-text2: #000000a3;
  --pixiv-logo-url: url('../images/logo.svg');
}

html,
body {
  color: #34495e;
}

section.cover {
  position: relative;
}

section.cover::after {
  position: absolute;
  top: 24px;
  left: 40px;

  /**
   * add alt text and fallback for non-supported browser
   * https://developer.mozilla.org/ja/docs/Web/CSS/content
   */
  content: var(--pixiv-logo-url);
  content: var(--pixiv-logo-url) / 'pixiv';
}

section.cover h1 {
  font-weight: bold;
}

section.cover ul {
  margin: 0 auto;
}

section.cover p {
  margin: 40px 0 0;
}

/* GitHub Button */
section.cover .cover-main > p:last-child a {
  font-weight: bold;
  border: 0;
  background-color: var(--color-surface3);
  color: var(--color-text2);
  margin: 0 8px;
}
