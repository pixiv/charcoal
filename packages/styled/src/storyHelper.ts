import { Material } from '@charcoal-ui/foundation'
import { CharcoalTheme } from '@charcoal-ui/theme'

export type MyTheme = CharcoalTheme & {
  color: {
    mycolor: Material
  }
}

export function myTheme(theme: CharcoalTheme): MyTheme {
  return {
    ...theme,
    color: {
      ...theme.color,
      mycolor: '#ff9e8c',
    },
  }
}
