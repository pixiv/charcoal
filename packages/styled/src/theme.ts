import { StyledTheme } from './types'
import {
  BorderRadius,
  Spacing,
  Typography,
  borderRadius,
  spacing,
  typography,
  Breakpoint,
  breakpoint,
} from '@charcoal/foundation'
import { light as l, dark as d } from '@charcoal/pixiv-theme'
import { Effect, Material, OpacityEffect, Theme } from '@charcoal/theme'
import { applyEffect } from '@charcoal/utils'

export interface ElementsTheme extends StyledTheme {
  color: Theme['color']
  gradientColor: Theme['gradientColor']
  effect: {
    hover: Effect
    press: Effect
  }
  elementEffect: {
    disabled: OpacityEffect
  }
  spacing: Spacing
  typography: {
    size: Typography
  }
  borderRadius: BorderRadius
  border: {
    default: {
      color: Material
    }
  }
  outline: {
    default: {
      color: string
      weight: 4
    }
    assertive: {
      color: string
      weight: 4
    }
  }
  breakpoint: Breakpoint
}

const outlineEffect = {
  type: 'opacity',
  opacity: 0.32,
} as const

const common = {
  borderRadius,
  spacing,
  typography: {
    size: typography,
  },
  breakpoint,
}

export const light: ElementsTheme = {
  color: l.color,
  gradientColor: l.gradientColor,
  effect: l.effect,
  elementEffect: l.elementEffect,
  border: {
    default: {
      color: l.color.border,
    },
  },
  outline: {
    default: {
      color: applyEffect(l.color.brand, outlineEffect),
      weight: 4,
    },
    assertive: {
      color: applyEffect(l.color.assertive, outlineEffect),
      weight: 4,
    },
  },
  ...common,
}

export const dark: ElementsTheme = {
  color: d.color,
  gradientColor: d.gradientColor,
  effect: d.effect,
  elementEffect: d.elementEffect,
  border: {
    default: {
      color: d.color.border,
    },
  },
  outline: {
    default: {
      color: applyEffect(d.color.brand, outlineEffect),
      weight: 4,
    },
    assertive: {
      color: applyEffect(d.color.assertive, outlineEffect),
      weight: 4,
    },
  },
  ...common,
}
