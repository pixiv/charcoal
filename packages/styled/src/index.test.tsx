import { light } from '@charcoal-ui/theme'
import 'jest-styled-components'

import React from 'react'
import renderder from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import { Example, TailwindLike } from './index.story'
import { myTheme } from './storyHelper'

function render(children: JSX.Element) {
  return renderder
    .create(<ThemeProvider theme={myTheme(light)}>{children}</ThemeProvider>)
    .toJSON()
}

// TODO: もう少し theme() 関数に対する丁寧なユニットテストが欲しい
describe('Story', () => {
  test('<Example />', () => {
    expect(render(<Example />)).toMatchSnapshot()
  })

  test('<TailwindLike />', () => {
    expect(render(<TailwindLike />)).toMatchSnapshot()
  })
})
