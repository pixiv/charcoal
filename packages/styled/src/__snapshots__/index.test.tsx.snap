// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Story <Example /> 1`] = `
.c1 {
  background-color: var(--charcoal-surface3);
  color: var(--charcoal-text4);
  font-size: 32px;
  line-height: 40px;
  font-family: monospace;
  display: flow-root;
  margin-top: 24px;
  margin-bottom: 24px;
  padding-top: 64px;
  padding-left: 64px;
}

.c1::before {
  display: block;
  width: 0;
  height: 0;
  content: '';
  margin-top: -4px;
}

.c1::after {
  display: block;
  width: 0;
  height: 0;
  content: '';
  margin-bottom: -4px;
}

.c0 {
  background-color: var(--charcoal-surface2);
  color: var(--charcoal-text2);
  font-size: 20px;
  line-height: 28px;
  font-weight: bold;
  padding-top: 36px;
  padding-bottom: 36px;
  padding-right: 40px;
  padding-left: 40px;
  margin-top: 24px;
  margin-bottom: 24px;
}

.c2 {
  background-color: var(--charcoal-surface2);
  padding-top: 40px;
  padding-bottom: 40px;
  padding-right: 40px;
  padding-left: 40px;
}

.c3 {
  background-color: var(--charcoal-surface4);
  color: var(--charcoal-text5);
  font-size: 20px;
  line-height: 28px;
  font-weight: bold;
  display: flow-root;
  height: 40px;
}

.c3::before {
  display: block;
  width: 0;
  height: 0;
  content: '';
  margin-top: -4px;
}

.c3::after {
  display: block;
  width: 0;
  height: 0;
  content: '';
  margin-bottom: -4px;
}

.c4 {
  background-color: var(--charcoal-surface3);
  color: var(--charcoal-text1);
  font-size: 16px;
  line-height: 24px;
  display: flow-root;
  width: 392px;
  margin-top: 24px;
  margin-bottom: 24px;
}

.c4::before {
  display: block;
  width: 0;
  height: 0;
  content: '';
  margin-top: -4px;
}

.c4::after {
  display: block;
  width: 0;
  height: 0;
  content: '';
  margin-bottom: -4px;
}

.c5 {
  height: 64px;
}

.c6 {
  background-color: var(--charcoal-surface4);
  color: var(--charcoal-text5);
  font-size: 20px;
  line-height: 28px;
  font-weight: bold;
  margin-top: 64px;
  margin-bottom: 24px;
  padding-top: 60px;
  padding-bottom: 60px;
  padding-right: 64px;
  padding-left: 64px;
  -webkit-transition: 0.2s color,0.2s background-color;
  transition: 0.2s color,0.2s background-color;
}

.c6:hover:not(:disabled):not([aria-disabled]),
.c6:hover[aria-disabled=false] {
  background-color: var(--charcoal-surface4-hover);
}

.c6:active:not(:disabled):not([aria-disabled]),
.c6:active[aria-disabled=false] {
  background-color: var(--charcoal-surface4-press);
}

.c6:hover:not(:disabled):not([aria-disabled]),
.c6:hover[aria-disabled=false] {
  color: var(--charcoal-text5-hover);
}

.c6:active:not(:disabled):not([aria-disabled]),
.c6:active[aria-disabled=false] {
  color: var(--charcoal-text5-press);
}

.c8 {
  background-color: var(--charcoal-surface3);
  font-size: 16px;
  line-height: 24px;
  font-weight: bold;
  display: flow-root;
  margin-top: 24px;
  margin-bottom: 24px;
  padding-top: 24px;
  padding-bottom: 24px;
  padding-right: 24px;
  padding-left: 24px;
}

.c8::before {
  display: block;
  width: 0;
  height: 0;
  content: '';
  margin-top: -4px;
}

.c8::after {
  display: block;
  width: 0;
  height: 0;
  content: '';
  margin-bottom: -4px;
}

.c7 {
  background-color: var(--charcoal-surface3);
  font-size: 16px;
  line-height: 24px;
  font-weight: bold;
  padding-top: 20px;
  padding-bottom: 20px;
  padding-right: 24px;
  padding-left: 24px;
  margin-top: 24px;
  margin-bottom: 24px;
}

.c9 {
  background-color: var(--charcoal-surface3);
  font-size: 16px;
  line-height: 24px;
  font-weight: bold;
  padding-top: 24px;
  padding-bottom: 24px;
  padding-right: 24px;
  padding-left: 24px;
  margin-top: 24px;
  margin-bottom: 24px;
}

.c10 {
  background-color: var(--charcoal-surface3);
  color: var(--charcoal-text2);
  font-size: 20px;
  line-height: 28px;
  font-weight: bold;
  display: flow-root;
  width: 288px;
  height: 64px;
  margin-top: 24px;
  margin-bottom: 24px;
}

.c10::before {
  display: block;
  width: 0;
  height: 0;
  content: '';
  margin-top: -4px;
}

.c10::after {
  display: block;
  width: 0;
  height: 0;
  content: '';
  margin-bottom: -4px;
}

.c11 {
  display: grid;
  grid-auto-flow: column;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  gap: 16px;
  margin-top: 24px;
  margin-bottom: 24px;
}

.c12 {
  font: inherit;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border-style: none;
  cursor: pointer;
  background-color: var(--charcoal-brand);
  color: var(--charcoal-text5);
  font-size: 14px;
  line-height: 22px;
  font-weight: bold;
  display: flow-root;
  height: 40px;
  padding-right: 24px;
  padding-left: 24px;
  border-radius: 999999px;
  -webkit-transition: 0.2s color,0.2s background-color,0.2s box-shadow;
  transition: 0.2s color,0.2s background-color,0.2s box-shadow;
}

.c12:focus {
  outline: none;
}

.c12:hover:not(:disabled):not([aria-disabled]),
.c12:hover[aria-disabled=false] {
  background-color: var(--charcoal-brand-hover);
}

.c12:active:not(:disabled):not([aria-disabled]),
.c12:active[aria-disabled=false] {
  background-color: var(--charcoal-brand-press);
}

.c12:hover:not(:disabled):not([aria-disabled]),
.c12:hover[aria-disabled=false] {
  color: var(--charcoal-text5-hover);
}

.c12:active:not(:disabled):not([aria-disabled]),
.c12:active[aria-disabled=false] {
  color: var(--charcoal-text5-press);
}

.c12::before {
  display: block;
  width: 0;
  height: 0;
  content: '';
  margin-top: -4px;
}

.c12::after {
  display: block;
  width: 0;
  height: 0;
  content: '';
  margin-bottom: -4px;
}

.c12:disabled,
.c12[aria-disabled]:not([aria-disabled=false]) {
  opacity: 0.32;
}

.c12:not(:disabled):not([aria-disabled]):focus,
.c12[aria-disabled=false]:focus,
.c12:not(:disabled):not([aria-disabled]):active,
.c12[aria-disabled=false]:active {
  outline: none;
  box-shadow: 0 0 0 4px rgba(0,150,250,0.32);
}

.c12:not(:disabled):not([aria-disabled]):focus:not(:focus-visible),
.c12[aria-disabled=false]:focus:not(:focus-visible),
.c12:not(:disabled):not([aria-disabled]):active:not(:focus-visible),
.c12[aria-disabled=false]:active:not(:focus-visible) {
  outline: none;
}

.c12:not(:disabled):not([aria-disabled]):focus-visible,
.c12[aria-disabled=false]:focus-visible {
  outline: none;
  box-shadow: 0 0 0 4px rgba(0,150,250,0.32);
}

.c12:disabled,
.c12[aria-disabled]:not([aria-disabled=false]) {
  cursor: unset;
}

.c13 {
  font: inherit;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border-style: none;
  height: 40px;
  width: 184px;
  padding-top: 0px;
  padding-bottom: 0px;
  padding-right: 8px;
  padding-left: 8px;
  border-radius: 4px;
  background-color: var(--charcoal-surface3);
  color: var(--charcoal-text2);
  font-size: 14px;
  line-height: 22px;
  -webkit-transition: 0.2s background-color,0.2s box-shadow;
  transition: 0.2s background-color,0.2s box-shadow;
}

.c13:focus {
  outline: none;
}

.c13:hover:not(:disabled):not([aria-disabled]),
.c13:hover[aria-disabled=false] {
  background-color: var(--charcoal-surface3-hover);
}

.c13:disabled,
.c13[aria-disabled]:not([aria-disabled=false]) {
  opacity: 0.32;
}

.c13:not(:disabled):not([aria-disabled]):focus,
.c13[aria-disabled=false]:focus,
.c13:not(:disabled):not([aria-disabled]):active,
.c13[aria-disabled=false]:active {
  outline: none;
  box-shadow: 0 0 0 4px rgba(0,150,250,0.32);
}

.c13:not(:disabled):not([aria-disabled]):focus:not(:focus-visible),
.c13[aria-disabled=false]:focus:not(:focus-visible),
.c13:not(:disabled):not([aria-disabled]):active:not(:focus-visible),
.c13[aria-disabled=false]:active:not(:focus-visible) {
  outline: none;
}

.c13:not(:disabled):not([aria-disabled]):focus-visible,
.c13[aria-disabled=false]:focus-visible {
  outline: none;
  box-shadow: 0 0 0 4px rgba(0,150,250,0.32);
}

.c14 {
  font: inherit;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border-style: none;
  height: 40px;
  width: 184px;
  padding-top: 0px;
  padding-bottom: 0px;
  padding-right: 8px;
  padding-left: 8px;
  border-radius: 4px;
  background-color: var(--charcoal-surface3);
  color: var(--charcoal-text2);
  font-size: 14px;
  line-height: 22px;
  -webkit-transition: 0.2s background-color,0.2s box-shadow;
  transition: 0.2s background-color,0.2s box-shadow;
}

.c14:focus {
  outline: none;
}

.c14:hover:not(:disabled):not([aria-disabled]),
.c14:hover[aria-disabled=false] {
  background-color: var(--charcoal-surface3-hover);
}

.c14:disabled,
.c14[aria-disabled]:not([aria-disabled=false]) {
  opacity: 0.32;
}

.c14:not(:disabled):not([aria-disabled]):focus,
.c14[aria-disabled=false]:focus,
.c14:not(:disabled):not([aria-disabled]):active,
.c14[aria-disabled=false]:active {
  outline: none;
  box-shadow: 0 0 0 4px rgba(0,150,250,0.32);
}

.c14:not(:disabled):not([aria-disabled]):focus:not(:focus-visible),
.c14[aria-disabled=false]:focus:not(:focus-visible),
.c14:not(:disabled):not([aria-disabled]):active:not(:focus-visible),
.c14[aria-disabled=false]:active:not(:focus-visible) {
  outline: none;
}

.c14:not(:disabled):not([aria-disabled]):focus-visible,
.c14[aria-disabled=false]:focus-visible {
  outline: none;
  box-shadow: 0 0 0 4px rgba(0,150,250,0.32);
}

.c14.c14:not(:disabled):not([aria-disabled]),
.c14.c14[aria-disabled=false] {
  box-shadow: 0 0 0 4px rgba(255,43,0,0.32);
}

.c15 {
  background-color: var(--charcoal-background1);
  color: var(--charcoal-text2);
  font-size: 14px;
  line-height: 22px;
  width: 392px;
  margin-top: 40px;
  margin-bottom: 40px;
  padding-top: 20px;
  padding-bottom: 20px;
  padding-right: 24px;
  padding-left: 24px;
  border: solid 1px rgba(0,0,0,0.08);
  border-radius: 8px;
}

.c16 {
  position: relative;
  z-index: 0;
  overflow: hidden;
  color: var(--charcoal-text5);
  width: 184px;
  padding-top: 36px;
  padding-bottom: 36px;
  padding-right: 40px;
  padding-left: 40px;
  margin-top: 24px;
  margin-bottom: 24px;
  font-size: 20px;
  line-height: 28px;
  font-weight: bold;
  border-radius: 8px;
  -webkit-transition: 0.2s color;
  transition: 0.2s color;
  text-align: center;
}

.c16::before {
  z-index: -1;
  content: '';
  display: block;
  position: absolute;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  -webkit-transition: 0.2s background-color;
  transition: 0.2s background-color;
}

.c16::after {
  z-index: -2;
  content: '';
  display: block;
  position: absolute;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  background-color: #d1ff1a;
  background-image: linear-gradient(to right,#d1ff1a 0%,#1ad1ff 100%);
}

.c16:hover:not(:disabled):not([aria-disabled])::before,
.c16:hover[aria-disabled=false]::before {
  background-color: rgba(0,0,0,0.04);
}

.c16:active:not(:disabled):not([aria-disabled])::before,
.c16:active[aria-disabled=false]::before {
  background-color: rgba(0,0,0,0.16);
}

.c16:hover:not(:disabled):not([aria-disabled]),
.c16:hover[aria-disabled=false] {
  color: var(--charcoal-text5-hover);
}

.c16:active:not(:disabled):not([aria-disabled]),
.c16:active[aria-disabled=false] {
  color: var(--charcoal-text5-press);
}

.c17 {
  background-color: #d1ff1a;
  background-image: linear-gradient(to right,#d1ff1a 0%,#1ad1ff 100%);
  color: var(--charcoal-text5);
  width: 184px;
  margin-top: 24px;
  margin-bottom: 24px;
  font-size: 20px;
  line-height: 28px;
  font-weight: bold;
  display: flow-root;
  border-radius: 8px;
  -webkit-transition: 0.2s color;
  transition: 0.2s color;
  padding-top: 40px;
  padding-bottom: 40px;
  padding-right: 40px;
  padding-left: 40px;
  text-align: center;
}

.c17:hover:not(:disabled):not([aria-disabled]),
.c17:hover[aria-disabled=false] {
  background-color: #c9f519;
  background-image: linear-gradient(to right,#c9f519 0%,#19c9f5 100%);
}

.c17:active:not(:disabled):not([aria-disabled]),
.c17:active[aria-disabled=false] {
  background-color: #b0d616;
  background-image: linear-gradient(to right,#b0d616 0%,#16b0d6 100%);
}

.c17:hover:not(:disabled):not([aria-disabled]),
.c17:hover[aria-disabled=false] {
  color: var(--charcoal-text5-hover);
}

.c17:active:not(:disabled):not([aria-disabled]),
.c17:active[aria-disabled=false] {
  color: var(--charcoal-text5-press);
}

.c17::before {
  display: block;
  width: 0;
  height: 0;
  content: '';
  margin-top: -4px;
}

.c17::after {
  display: block;
  width: 0;
  height: 0;
  content: '';
  margin-bottom: -4px;
}

.c18 {
  width: 288px;
  margin-top: 24px;
  margin-bottom: 24px;
}

.c19 {
  display: inline-block;
  width: 100%;
  background-color: var(--charcoal-surface4);
  color: var(--charcoal-text5);
  padding-top: 16px;
  padding-bottom: 16px;
  padding-right: 16px;
  padding-left: 16px;
}

.c20 {
  --charcoal-text1: #ffff00;
  --charcoal-text1-hover: #f5f500;
  --charcoal-text1-press: #d6d600;
  background-color: var(--charcoal-surface4);
  color: var(--charcoal-text1);
}

<div
  className=""
>
  <div
    className="c0"
  >
    Sample
  </div>
  <div
    className="c1"
  >
    Left Top Padding
  </div>
  <div
    className="c2"
  >
    <div
      className="c3"
    >
      Nested
    </div>
  </div>
  <div
    className="c4"
  >
    Lorem ipsum, dolor sit amet consectetur adipisicing elit. Autem dolores, recusandae quidem mollitia eum non vel architecto possimus repudiandae quis molestias neque facilis rem dolorum voluptatem impedit nemo praesentium voluptas.
  </div>
  <div
    className="c4 c5"
  >
    Lorem ipsum, dolor sit amet consectetur adipisicing elit. Autem dolores, recusandae quidem mollitia eum non vel architecto possimus repudiandae quis molestias neque facilis rem dolorum voluptatem impedit nemo praesentium voluptas.
  </div>
  <div
    className="c6"
  >
    With effects
  </div>
  <div
    className="c7"
  >
    Without half-leading
  </div>
  <div
    className="c8"
  >
    Without half-leading (No optimization)
  </div>
  <div
    className="c9"
  >
    With half-leading
  </div>
  <div
    className="c10"
  >
    Fixed size box
  </div>
  <div
    className="c11"
  >
    <button
      className="c12"
    >
      Button
    </button>
    <button
      className="c12"
      disabled={true}
    >
      Disabled
    </button>
  </div>
  <div
    className="c11"
  >
    <input
      className="c13"
      type="text"
      value="text field"
    />
    <input
      className="c13"
      disabled={true}
      type="text"
      value="disabled"
    />
    <input
      className="c14"
      type="text"
      value="invalid"
    />
  </div>
  <div
    className="c15"
  >
    Border
  </div>
  <div
    className="c16"
  >
    Gradient
  </div>
  <div
    className="c17"
  >
    Gradient (Warning)
  </div>
  <div
    css="
      display: flex;
    "
  >
    <div
      css={[Function]}
    >
      Tailwind like
    </div>
  </div>
  <div
    className="c18"
  >
    <div
      className="c19"
    >
      Full width
    </div>
  </div>
  <div
    className="c20"
  >
    This is actually text1 !
  </div>
</div>
`;

exports[`Story <TailwindLike /> 1`] = `
<div
  css="
      display: flex;
    "
>
  <div
    css={[Function]}
  >
    Tailwind like
  </div>
</div>
`;
